### ISSUE TITLE
[Bug] Openvino model converted from TF 2 EfficientDet outputs wrong class index of detection.

### ISSUE THREAD

**Issue by fangliang425** [openvinotoolkit/openvino#7000]
##### System information (version)
- OpenVINO => 2021.4
- Operating System / Platform => Ubuntu 20.04
- Problem classification: Model Conversion
- Framework: Tensorflow 2.4
- Model name: [EfficientDet D0 512x512](http://download.tensorflow.org/models/object_detection/tf2/20200711/efficientdet_d0_coco17_tpu-32.tar.gz)
- Model optimizer conversion file: <INSTALL_DIR>/deployment_tools/model_optimizer/extensions/front/tf/efficient_det_support_api_v2.0.json

##### Detailed description
Model conversion from tensorflow saved model works fine with the tutorial,[https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html)

and we also managed to run the openvino model. However, the class index of detection from openvino model is always 1 less than tensorflow’s. We tested three pictures by running original tensorflow saved model and openvino, in which the red label is class index of COCO dataset and the blue one is the score of detection. 

As you can see, boxes and scores from two models are exactly the same while class indices have differece of 1.
Particularly for the second and third pictures, persons should be detected as class 1, but they are detected as class 0 (background) and discarded due to the bug.
[here to check COCO classes](https://tech.amikelive.com/node-718/what-object-categories-labels-are-in-coco-dataset/)

![image1](https://user-images.githubusercontent.com/10607604/128833962-03273a4c-d2d8-4e04-9bec-4c760ca82c41.jpg)
![image2](https://user-images.githubusercontent.com/10607604/128833987-fc1c6ce5-af2e-4999-9e4b-bfdf53313236.jpg)
![image4](https://user-images.githubusercontent.com/10607604/128834001-36424b06-dcd6-48b6-a7c0-79ed9cb7aab6.jpg)

##### Steps to reproduce
- Download the Tensorflow model [EfficientDet D0 512x512](http://download.tensorflow.org/models/object_detection/tf2/20200711/efficientdet_d0_coco17_tpu-32.tar.gz).
- Download the [openvino model](https://github.com/fangliang425/tmp/raw/main/EfficientDet_Openvino.tar.gz) converted from EfficientDet savedmodel.
- Download [test images.](https://github.com/fangliang425/tmp/raw/main/test_images.tar.gz)
- Running [Tenorflow demo code](https://github.com/fangliang425/tmp/blob/main/demo_code/EfficientDet_Tensorflow_test.py) and [openvino demo code](https://github.com/fangliang425/tmp/blob/main/demo_code/EfficientDet_OpenVINO_test.py).

**Comment by Iffa-Meah** [comment]
I had tested the model (xml and saved model) and I get the same result as yours.
You may refer to my attachment.

I noticed that for the OpenVINO demo, you are using IR format which means you had converted the model using **model optimizer**.
Meanwhile, for the Tensorflow demo, you are using saved model format which are used as it is when you download the model.

The model optimizer convert a model to IR (xml,bin) and at the same time it optimized the model by using various techniques (fusion of layers,etc). Plus, it provides a scalability in term of weight format (FP32,FP16,etc).

Probably the optimization is what causing your index reading to differ by 1.

You may refer [here](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimization_Techniques.html).


![test](https://user-images.githubusercontent.com/79643476/129119826-a56281a1-abd0-40cf-90c2-db92e10fc7f0.png)

**Comment by fangliang425** [comment]
Thanks for reply, so any solutions for this issue?
Conversion command as below, which is from your [tutorial](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html).

`python <INSTALL_DIR>/deployment_tools/model_optimizer/mo_tf.py --saved_model_dir <PATH>/efficientdet_d0_coco17_tpu-32/saved_model/ --transformations_config <INSTALL_DIR>/deployment_tools/model_optimizer/extensions/front/tf/efficient_det_support_api_v2.0.json --tensorflow_object_detection_api_pipeline_config <PATH>/efficientdet_d0_coco17_tpu-32/pipeline.config --output_dir <OUTPUT_PATH>`

**Comment by Iffa-Meah** [comment]
It is recommended for you to use the model officially downloaded from OpenVINO model downloader and run it with Official OpenVINO object detection python demo. However with this, you can only use the IR format.

From my end, I had checked both FP32 and FP16. They both give out the same result.

1. Download
<openvino path>\deployment_tools\tools\model_downloader\downloader.py" --name efficientdet-d0-tf

2. Convert
<openvino path>\deployment_tools\tools\model_downloader\converter.py" --name efficientdet-d0-tf

3. Run with demo
<openvino path>\inference_engine\demos\object_detection_demo\python\object_detection_demo.py" --model efficientdet-d0-tf.xml --input image4.jpg -at ssd --loop

**Comment by fangliang425** [comment]
The COCO pre-trained model I posted is just to demonstrate the issue for you, we do need to train the tf efficientdet with our own data. It doesn't make sense to use the official openvino model.

So, there is something wrong with the conversion of EfficientDet, and you don't have the solution yet so far, right?

**Comment by Iffa-Meah** [comment]
Closing issue, feel free to re-open or start a new issue if additional assistance is needed.
